catalog/spatial/registry/registry-federation-admin-impl/src/main/java/org/codice/ddf/registry/federationadmin/impl/FederationAdmin.java

```diff
@@ -0,0 +1,699 @@
+        }
+        Metacard existingMetacard = existingMetacards.get(0);
+
+        Metacard updateMetacard = getRegistryMetacardFromRegistryPackage(registryPackage);
+        updateMetacard.setAttribute(new AttributeImpl(Metacard.ID, existingMetacard.getId()));
+
+        federationAdminService.updateRegistryEntry(updateMetacard);
+    }
+
+    @Override
+    public void deleteLocalEntry(List<String> ids) throws FederationAdminException {
+        if (CollectionUtils.isEmpty(ids)) {
+            throw new FederationAdminException(
+                    "Error deleting local registry entries. No ids provided.");
+        }
+
+        List<Metacard> localMetacards =
+                federationAdminService.getLocalRegistryMetacardsByRegistryIds(ids);
+        List<String> metacardIds = new ArrayList<>();
+
+        metacardIds.addAll(localMetacards.stream()
+                .map(Metacard::getId)
+                .collect(Collectors.toList()));
+        if (ids.size() != metacardIds.size()) {
+            String message = "Error deleting local registry entries. Registry ids provided.";
+            LOGGER.error("{} Registry Ids provided: {}. Registry metacard ids found: {}",
+                    message,
+                    ids,
+                    metacardIds);
+            throw new FederationAdminException(message);
+        }
+
+        federationAdminService.deleteRegistryEntriesByMetacardIds(metacardIds);
+    }
+
+    @Override
+    public Map<String, Object> getLocalNodes() throws FederationAdminException {
+        Map<String, Object> localNodes = new HashMap<>();
+        List<Map<String, Object>> registryWebMaps = new ArrayList<>();
+
+        List<RegistryPackageType> registryPackages =
+                federationAdminService.getLocalRegistryObjects();
+
+        List<Metacard> metacards = federationAdminService.getLocalRegistryMetacards();
+        Map<String, Metacard> metacardByRegistryIdMap = getRegistryIdMetacardMap(metacards);
+
+        registryWebMaps = getWebMapsFromRegistryPackages(registryPackages, metacardByRegistryIdMap);
+
+        if (customSlots != null) {
+            localNodes.put(CUSTOM_SLOTS_KEY, customSlots);
+        }
+
+        Map<String, Object> autoPopulateMap = new HashMap<>();
+        autoPopulateMap.put(SERVICE_BINDINGS_KEY, endpointMap.values());
+        localNodes.put(AUTO_POPULATE_VALUES_KEY, autoPopulateMap);
+
+        localNodes.put(NODES_KEY, registryWebMaps);
+
+        return localNodes;
+    }
+
+    @Override
+    public boolean registryStatus(String servicePID) {
+        try {
+            List<Source> sources = helper.getRegistrySources();
+            for (Source source : sources) {
+                if (source instanceof ConfiguredService) {
+                    ConfiguredService cs = (ConfiguredService) source;
+                    try {
+                        Configuration config = helper.getConfiguration(cs);
+                        if (config != null && config.getProperties()
```

Comment:
Could use this constant: https://osgi.org/javadoc/r2/org/osgi/framework/Constants.html#SERVICE_PID